app/src/main/java/nl/tudelft/pixelperfect/client/ClientListener.java

```diff
@@ -28,19 +32,34 @@ public void messageReceived(Client source, Message message) {
             switch (eve.getType()) {
                 case "FireEvent":
                     mission = new FireEvent(eve.getID(), "", "", eve.getTime(), eve.getDuration(), 0);
-                    GameActivity.updateEventLog(mission);
+                    LocationArmoryActivity.updateEventLog(mission);
                     break;
                 case "AsteroidFieldEvent":
                     mission = new AsteroidFieldEvent(eve.getID(), "", "", eve.getTime(), eve.getDuration(), 0);
-                    GameActivity.updateEventLog(mission);
+                    LocationLabActivity.updateEventLog(mission);
                     break;
                 case "HostileShipEvent":
                     mission = new HostileShipEvent(eve.getID(), "", "", eve.getTime(), eve.getDuration(), 0);
-                    GameActivity.updateEventLog(mission);
+                    LocationArmoryActivity.updateEventLog(mission);
                     break;
                 case "PlasmaLeakEvent":
                     mission = new PlasmaLeakEvent(eve.getID(), "", "", eve.getTime(), eve.getDuration(), 0);
-                    GameActivity.updateEventLog(mission);
+                    LocationEngineroomActivity.updateEventLog(mission);
+                    break;
+                default:
+                    break;
+            }
+        } else if (message instanceof RoleChosenMessage) {
+            RoleChosenMessage roleMessage = (RoleChosenMessage) message;
+            switch(roleMessage.getRole()) {
+                case GUNNER :
+                    RoleActivity.updateButtons(Roles.GUNNER);
```

Comment:
SUGGESTION: while this is good enough (imo), consider moving the updateButton method to each respective Enum and just call roleMessage.getRole().updateButton() to avoid the switch all together.